### PR TITLE
Add autoclose dm config parameter

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
Adds a config parameter.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7984

#### Checklist
N/A

### CODE REVIEW

File: model/config.go [hunk 1]
```diff
@@ -204,6 +204,7 @@ type ServiceSettings struct {
 	EnableChannelViewedMessages              *bool
 	EnableUserStatuses                       *bool
 	ClusterLogTimeoutMilliseconds            *int
+	CloseUnusedDirectMessagesInSidebar       *bool
```
Comment (enahum) [inline model/config.go]: 0/5 here, but maybe we can remove `InSidebar` as that only refers to the webapp

Comment (ccbrown) [reply model/config.go]: Completely agree. The JIRA ticket was just very specific about the name (It wasn't my choosing.).

@esethna I assume it's okay if we deviate from that a bit?

Comment (esethna) [reply model/config.go]: Yes that's fine @ccbrown, we can go with "CloseUnusedDirectMessages"

**Comment by ccbrown** [comment]
@enahum Updated. If all looks good to you, I think we can go ahead and merge this one.

**Pull request merged**